Use Python Script Instead of Bash Scripts

ryan-clancy opened the pull request:
@lintool Here's a first stab at the Python script implementation...

Instead of running the bash scripts, you just make sure the `config.json` file points to the index and config file locations for your collection, run the script, and reload the cores from the admin UI.

Commit:
Initial implementation of the Python script.

- Removed the load.sh and run.sh scripts
- Add run.py which reads config.json to build the Docker image based on the indexes specified in the config.

Commit:
Add comments

Commit:
Update README.md

lintool commented on config.json:
```diff
@@ -0,0 +1,17 @@
+    {
+      "name": "core17",
+      "index_path": "/home/ryan/waterloo/projects/index/lucene-index.core17.pos+docvectors+rawdocs",
+      "config_path": "/home/ryan/waterloo/projects/SolrAnserini/configsets/core17"
```
Change this to a relative path, relative to current directory, e.g., `configsets/core17`

ryan-clancy replied:
Added in latest commits.

lintool commented on config.json:
```diff
@@ -0,0 +1,17 @@
+  "indexes": [
+    {
+      "name": "core17",
+      "index_path": "/home/ryan/waterloo/projects/index/lucene-index.core17.pos+docvectors+rawdocs",
```
Change this to a relative path? Defaults shouldn't have tour home directory hard coded?

ryan-clancy replied:
Added in latest commits.

lintool commented on run.py:
```diff
@@ -0,0 +1,101 @@
+                                 volumes=volumes)
+
+
+if __name__ == '__main__':
```
Shouldn't the config file be passed in as a config parameter? E.g., `python run.py -config config.json`?

ryan-clancy replied:
Added in latest commits.

Commit:
Use relative path for index and config files.

Also use a better name for the "name" config option and remove the
"config_mount" option (it's static in Solr).

Commit:
Add flag for specifying config location

Commit:
Add .gitignore

Commit:
Merge branch 'python-script' of github.com:castorini/SolrAnserini into python-script

Commit:
Update README.md...

and add help for command line args

paopao74cn commented on run.py:
```diff
@@ -0,0 +1,107 @@
+import os
+import time
+
+import docker
```
This requires installing Docker SDK for Python:
```
pip install docker
```

ryan-clancy replied:
Thanks for the catch! Added that to the README

paopao74cn reviewed (approved):
Just tested. It is working fine! 👍

Commit:
Update README.md

paopao74cn reviewed (approved):
I think it is good to go.
